test/extended/cli/mustgather.go

```diff
@@ -285,6 +290,59 @@ var _ = g.Describe("[sig-cli] oc adm must-gather", func() {
 			o.Expect(fmt.Errorf("expected files should not be empty: %s", strings.Join(emptyFiles, ","))).NotTo(o.HaveOccurred())
 		}
 	})
+
+	g.When("looking at the audit logs", func() {
+		g.Describe("[sig-node] kubelet", func() {
+			g.It("runs apiserver processes strictly sequentially in order to not risk audit log corruption", func() {
+				// On IBM ROKS, events will not be part of the output, since audit logs do not include control plane logs.
+				if e2e.TestContext.Provider == ibmcloud.ProviderName {
+					g.Skip("ROKs doesn't have audit logs")
+				}
+
+				tempDir, err := ioutil.TempDir("", "test.oc-adm-must-gather.")
+				o.Expect(err).NotTo(o.HaveOccurred())
+				// defer os.RemoveAll(tempDir)
+
+				args := []string{
+					"--dest-dir", tempDir,
+					"--",
+					"/usr/bin/gather_audit_logs",
+				}
+
+				o.Expect(oc.Run("adm", "must-gather").Args(args...).Execute()).To(o.Succeed())
+
+				pluginOutputDir := getPluginOutputDir(oc, tempDir)
+				expectedAuditSubDirs := []string{"kube-apiserver", "openshift-apiserver", "oauth-apiserver"}
+
+				seen := sets.String{}
+				for _, apiserver := range expectedAuditSubDirs {
+					err := filepath.Walk(filepath.Join(pluginOutputDir, "audit_logs", apiserver), func(path string, info os.FileInfo, err error) error {
+						g.By(path)
+						o.Expect(err).NotTo(o.HaveOccurred())
+						if info.IsDir() {
+							return nil
+						}
+
+						seen.Insert(apiserver)
+
+						if path != "lock.log" {
```

Comment:
Shouldn't we extract the file name from the path?
Otherwise this condition will always be true since the path doesn't contain only the file name.